browser/ui/tabs/split_view_tab_strip_model_adapter.cc

```diff
@@ -52,6 +53,29 @@ void SplitViewTabStripModelAdapter::TabDragStarted() {
   is_in_tab_dragging_ = true;
 }
 
+void SplitViewTabStripModelAdapter::SynchronizeGroupedState(
+    const SplitViewBrowserData::Tile& tile,
+    const tabs::TabHandle& source,
+    std::optional<tab_groups::TabGroupId> group) {
+  DCHECK(!is_in_synch_grouped_state_);
+  base::AutoReset<bool> resetter(&is_in_synch_grouped_state_, true);
+  auto other_tab = tile.first == source ? tile.second : tile.first;
+  auto other_tab_index = model_->GetIndexOfTab(other_tab);
+  auto tab_group_for_secondary_tab = model_->GetTabGroupForTab(other_tab_index);
+  if (group == tab_group_for_secondary_tab) {
+    return;
+  }
+
+  auto tab_index = model_->GetIndexOfTab(other_tab);
+  if (group) {
+    model_->AddToExistingGroup({tab_index}, group.value());
+  } else {
+    model_->RemoveFromGroup({tab_index});
+  }
+
+  MakeTiledTabsAdjacent(tile, true);
```

Comment:
It seems redundant call as `SplitViewBrowserData::TileTabs()` also call `MarkTiledTabsAdjacent()`?

Reply:
Good point. Changed callsite.